android/expoview/src/main/java/host/exp/exponent/utils/ToastHelper.kt

```diff
@@ -0,0 +1,38 @@
+package host.exp.exponent.utils
+
+import android.app.Application
+import android.os.Build
+import android.view.Gravity
+import android.widget.Toast
+import host.exp.exponent.di.NativeModuleDepsProvider
+import javax.inject.Inject
+
+// TODO: Remove when we drop support for SDK 40
+
+object ToastHelper {
+  @Inject
+  var applicationContext: Application? = null
+
+  init {
+    NativeModuleDepsProvider.getInstance().inject(ToastHelper::class.java, this)
+  }
+
+  fun functionMayNotWorkOnAndroidRWarning() {
+    try {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        applicationContext?.let {
+          val message = "This feature may not work in Expo Go when you're using Android R.\nSee https://expo.fyi/android-r"
```

Comment:
we should make this function accept a string that indicates what feature we are referring to - "Saving assets to albums with expo-media-library may not work as expected" and "Reading the device GPS location with expo-location may not work as expected" instead of just "This feature may not work" :)